tensor2tensor/data_generators/text_encoder.py

```diff
@@ -393,8 +393,8 @@ def store_to_file(self, filename):
       filename: Full path of the file to store the vocab to.
     """
     with tf.gfile.Open(filename, "w") as f:
-      for i in range(len(self._id_to_token)):
-        f.write(self._id_to_token[i] + "\n")
+      for token in self._id_to_token:
+        f.write(token + "\n")
```

Comment:
this will not work, _id_to_token is a dict and the previous code basically writes tghe vocab file in order.

Reply:
Good catch. I missed the fact that dicts aren't order prior to Python 3.6.